arch/arm/core/aarch32/swap_helper.S

```diff
@@ -51,10 +51,10 @@ GDATA(_kernel)
 
 SECTION_FUNC(TEXT, z_arm_pendsv)
 
-#ifdef CONFIG_TRACING
+#if defined(CONFIG_TRACING) || defined(CONFIG_THREAD_RUNTIME_STATS)
```

Comment:
we could get rid of this #if here and everwhere else where this function is called and move things to where the function is implemented. It would be an empty function if no tracing or runtime stats are configured, leaves us with less ifdefs and and someone else decides to use this hook, there will be only once place to change.

Reply:
My understanding is that an empty function in another file would still get called. So here we need to do push/pop even when tracing or runtime stats are not enabled. I can add an indirect kconfig (where the others would select this one) to replace the current #ifdef. So we don't need to branch/jump unless necessary.

Reply:
fair

Reply:
Replaced with an indirect kconfig.